src/solvers/dg_tree/dg_2d.jl

```diff
@@ -182,14 +186,12 @@ function calc_volume_integral!(du, u,
 
       flux1 = flux(u_node, 1, equations)
       for ii in eachnode(dg)
-        integral_contribution = derivative_dhat[ii, i] * flux1
-        add_to_node_vars!(du, integral_contribution, equations, dg, ii, j, element)
+        add_to_node_vars!(du, derivative_dhat[ii, i], flux1, equations, dg, ii, j, element)
       end
 
       flux2 = flux(u_node, 2, equations)
       for jj in eachnode(dg)
-        integral_contribution = derivative_dhat[jj, j] * flux2
-        add_to_node_vars!(du, integral_contribution, equations, dg, i, jj, element)
+        add_to_node_vars!(du, derivative_dhat[jj, j], flux2, equations, dg, i, jj, element)
```

Comment:
For the same reasons as in my comment on `inv_γm1`, I would argue that this is a performance optimization that is not strictly necessary while making the code harder to read and understand. However, in this case it is already not easy to know what's going on, thus my feelings are less strong. Maybe rename this function to `multiply_and_add_to_node_vars!` (or similar)?

Reply:
I can foresee that without a name change like @sloede suggested this will be very confusing to new users. To the untrained eye it looks like we are adding the `jj, j` entry of the derivative matrix and `flux2` into the `du` variable for some reason. It is very obscure that a product is being done internally in the function `add_to_node_vars`

Reply:
What about `muladd_to_node_vars!`?

Reply:
I agree that my original suggestions was a tad too long. However, also `muladd_...` doesn't really vibe with me, as it is IMHO only self-descriptive if you already aware of this FMA discussion. How about `multiply_add_to_node_vars!`?

Reply:
Well, `muladd` is a Julia Base function... But anyway, `multiply_add_to_node_vars!` is fine for me as well